context.go

```diff
@@ -87,8 +88,14 @@ func defaultGOPATH() string {
 // SourceManager produces an instance of gps's built-in SourceManager
 // initialized to log to the receiver's logger.
 func (c *Ctx) SourceManager() (*gps.SourceMgr, error) {
+	cachedir := c.Cachedir
+	if cachedir == "" {
+		// When `DEPCACHEDIR` isn't set in the env, fallback to `$GOPATH/pkg/dep`.
+		cachedir = filepath.Join(c.GOPATH, "pkg", "dep")
+	}
+
```

Comment:
This is probably where where first the `os.Mkdir()` call should be made to attempt to ensure that the `cachedir` exists prior to setting up the `SourceManager`, so that the calls within `NewSourceManager()` won't have a problem.